### PR TITLE
JdbcIO.readRows() to handle DECIMAL(10,2) with scale as the proper LogicalType.

### PR DESCRIPTION
**Pull request by mkhludnev**
When JdbcIO.readRows() schema ignores scale in DECIMAL(20,2) 
This is the first of the problem described at https://lists.apache.org/thread/qxwron01myf5fzrb3d8tsh71grt5cpzk

### CODE REVIEW

**Comment by mkhludnev** [comment]
This is the first of the problem described at https://lists.apache.org/thread/qxwron01myf5fzrb3d8tsh71grt5cpzk

File: sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java [hunk 1]
```diff
@@ -64,15 +64,17 @@ public void testToBeamSchema() throws SQLException {
             JdbcFieldInfo.of("boolean_col", Types.BOOLEAN),
             JdbcFieldInfo.of("char_col", Types.CHAR, 255),
             JdbcFieldInfo.of("date_col", Types.DATE),
-            JdbcFieldInfo.of("decimal_col", Types.DECIMAL),
+            JdbcFieldInfo.of("decimal_col", Types.DECIMAL, Integer.MAX_VALUE),
```
Comment (aromanenko-dev) [inline sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java:67]: Why not keep just `JdbcFieldInfo.of("decimal_col", Types.DECIMAL)` for this?

**Review by aromanenko-dev — APPROVED** [review]
Thanks, LGTM
Just one minor comment and, please, add a Github issue for this (if not yet) or link it with existing one.

File: sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java (hunk 1 above)
Comment (mkhludnev) [reply sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java:67]: but it means DECIMAL(0,0) zero precision, zero scale. Despite it's possible in Java, it's not possible in real database, I believe. eg. If I attempt to declare such column in Postgres I've got `NUMERIC precision 0 must be between 1 and 1000`. So, I fixed this stub to mimic real DB metadata.

**Comment by mkhludnev** [comment]
fixes #27380

**Comment by mkhludnev** [comment]
> Thanks, LGTM Just one minor comment and, please, add a Github issue for this (if not yet) or link it with existing one.

Here we go #27380

**Comment by Abacn** [comment]
Run Python 3.8 PostCommit

**Comment by Abacn** [comment]
just check the if the change affecting python jdbcio wrapper / need change in py side

File: sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java [hunk 2]
```diff
@@ -105,7 +109,9 @@ public void testToBeamSchema() throws SQLException {
             .addField("boolean_col", Schema.FieldType.BOOLEAN)
             .addField("char_col", LogicalTypes.fixedLengthString(JDBCType.CHAR, 255))
             .addField("date_col", LogicalTypes.JDBC_DATE_TYPE)
-            .addField("decimal_col", Schema.FieldType.DECIMAL)
+            .addField("decimal_col", LogicalTypes.numeric(0, 0)) // doubtful
```
Comment (mkhludnev) [inline sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java:112]: even it's not degradation, it's really suspicious. Should we add `precision==0` as magic condition into `SchemaUtil. beamLogicalNumericField ()`?

File: sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java (hunk 1 above)
Comment (aromanenko-dev) [reply sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java]: If it declares `DECIMAL(0,0)`, won't it uses just default precision and scale in this case?

**Comment by aromanenko-dev** [comment]
> Here we go #27380
Btw, do you have a regression test for this case (`DECIMAL(10,2)`) that fails without your changes and not fail with?

**Comment by Abacn** [comment]
Run Python 3.8 PostCommit

**Comment by Abacn** [comment]
at least on MySQL and Postgres, `DECIMAL(10,2)` works for readRows, from this integration test (python wrapper): https://github.com/apache/beam/blob/a5873792a36507de6909580b8c4867e09c49d8bb/sdks/python/apache_beam/io/external/xlang_jdbcio_it_test.py#L135

**Comment by Abacn** [comment]
Run Python 3.8 PostCommit

File: CHANGES.md [hunk 3]
```diff
@@ -80,6 +80,7 @@
 
 * Fixed X (Java/Python) ([#X](https://github.com/apache/beam/issues/X)).
 * Fixed KinesisIO `NullPointerException` when a progress check is made before the reader is started (IO) ([#23868](https://github.com/apache/beam/issues/23868))
+* Fixed Schema representation for DECIMAL(10,2) with precision and scale (JdbcIO) ([#27380](https://github.com/apache/beam/issues/27380))
```
Comment (Abacn) [inline CHANGES.md:83]: This goes to v2.50.0

**Comment by mkhludnev** [comment]
> > Here we go #27380
> 
> Btw, do you have a regression test for this case (`DECIMAL(10,2)`) that fails without your changes and not fail with?

ok. Let me step back and come up with integration test. 

@Abacn 
Thanks for your notes I'll check it out.

**Review by aromanenko-dev — CHANGES_REQUESTED** [review]
After discussion, it was decided to double-check this issue and fix again by author.

**Comment by mkhludnev** [comment]
> > Here we go #27380
> 
> Btw, do you have a regression test for this case (`DECIMAL(10,2)`) that fails without your changes and not fail with?

#27406 here we go. It's either really upset, or I'm missing the point completely.

**Comment by mkhludnev** [comment]
tbc #27380

File: sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java (hunk 1 above)
Comment (mkhludnev) [reply sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/SchemaUtilTest.java]: > If it declares `DECIMAL(0,0)`, won't it uses just default precision and scale in this case?

https://github.com/apache/beam/pull/27406#issuecomment-1630313700
From what I read https://www.postgresql.org/docs/current/datatype-numeric.html 0 precision means variable number of digits.

**Pull request closed without merging**